ben/public/scripts/article.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+var app = app || {};
+
+(function(module){
+  function Article(rawDataObj) {
+    // REVIEW: In Lab 8, we explored a lot of new functionality going on here. Let's re-examine the concept of context. Normally, "this" inside of a constructor function refers to the newly instantiated object. However, in the function we're passing to forEach, "this" would normally refer to "undefined" in strict mode. As a result, we had to pass a second argument to forEach to make sure our "this" was still referring to our instantiated object. One of the primary purposes of lexical arrow functions, besides cleaning up syntax to use fewer lines of code, is to also preserve context. That means that when you declare a function using lexical arrows, "this" inside the function will still be the same "this" as it was outside the function. As a result, we no longer have to pass in the optional "this" argument to forEach!
+    Object.keys(rawDataObj).forEach(key => this[key] = rawDataObj[key]);
+  }
+
+  Article.prototype.toHtml = function() {
+    var template = Handlebars.compile($('#article-template').text());
+
+    this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+    this.publishStatus = this.publishedOn ? `published ${this.daysAgo} days ago` : '(draft)';
+    this.body = marked(this.body);
+
+    return template(this);
+  };
+
+  Article.loadAll = articleData => {
+    articleData.sort((a,b) => (new Date(b.publishedOn)) - (new Date(a.publishedOn)))
+
+    /* OLD forEach():
+    articleData.forEach(articleObject => Article.all.push(new Article(articleObject)));
+    */
+    Article.all = articleData.map(articleObject => new Article(articleObject));
+    console.log(Article.all);
+  };
+
+  Article.fetchAll = callback => {
+    $.get('/articles')
+      .then(results => {
+        Article.loadAll(results);
+        callback();
+      })
+  };
+
+  // Hint: What property of an individual instance contains the main text of the article?
+  Article.numWordsAll = () => {
+    return Article.all.map(article => article.body.split(" ").length).reduce((acc, cur) => acc + cur)
+  };
+
+  // Hint: Make sure to return an array and avoid duplicates.
+  Article.allAuthors = () => {
+    return Article.all.map(article => article.author).reduce((acc, cur) => {
+      acc.includes(cur) ? null : acc.push(cur);
```

Comment:
I'd rather do this as a conditional than a ternary, since its goal isn't to get out one value.
i.e.
```
if (!acc.includes(cur)) {
  acc.push(cur);
}
return acc;
```